app/payment-link-v2/reference/get-index.controller.js

```diff
@@ -0,0 +1,32 @@
+'use strict'
+
+const { response } = require('../../utils/response')
+const { getSessionVariable } = require('../../utils/cookie')
+const { NotFoundError } = require('../../errors')
+const getBackLinkUrl = require('./get-back-link-url')
+
+module.exports = function getReferencePage (req, res, next) {
+  const product = req.product
+
+  const data = {
+    productExternalId: product.externalId,
+    productName: product.name
+  }
+
+  if (!product.reference_enabled) {
+    return next(new NotFoundError('Attempted to access reference page with a product that auto-generates references.'))
+  }
+
+  const sessionReferenceNumber = getSessionVariable(req, 'referenceNumber')
+
+  data.backLinkHref = getBackLinkUrl(sessionReferenceNumber, product)
+
+  if (sessionReferenceNumber) {
+    data.referenceNumber = sessionReferenceNumber
+  }
+
+  data.paymentReferenceLabel = product.reference_label
+  data.paymentReferenceHint = product.reference_hint
```

Comment:
Could probably move this up to where we instantiate `data` just to make things a little cleaner.

Reply:
Good idea - done